src/test/java/com/synopsys/integration/alert/database/job/DistributionJobRepositoryTestIT.java

```diff
@@ -0,0 +1,336 @@
+package com.synopsys.integration.alert.database.job;
+
+import com.synopsys.integration.alert.common.enumeration.FrequencyType;
+import com.synopsys.integration.alert.common.enumeration.ProcessingType;
+import com.synopsys.integration.alert.common.persistence.model.job.DistributionJobModel;
+import com.synopsys.integration.alert.common.persistence.model.job.DistributionJobRequestModel;
+import com.synopsys.integration.alert.common.persistence.model.job.details.DistributionJobDetailsModel;
+import com.synopsys.integration.alert.common.persistence.model.job.details.EmailJobDetailsModel;
+import com.synopsys.integration.alert.common.rest.model.AlertNotificationModel;
+import com.synopsys.integration.alert.common.util.DateUtils;
+import com.synopsys.integration.alert.database.api.DefaultNotificationAccessor;
+import com.synopsys.integration.alert.database.api.StaticJobAccessor;
+import com.synopsys.integration.alert.descriptor.api.model.ChannelKeys;
+import com.synopsys.integration.alert.util.AlertIntegrationTest;
+import com.synopsys.integration.blackduck.api.generated.enumeration.VulnerabilitySeverityType;
+import com.synopsys.integration.blackduck.api.manual.enumeration.NotificationType;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.OffsetDateTime;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@AlertIntegrationTest
+class DistributionJobRepositoryTestIT {
+    // Distribution jobs related
+    @Autowired private DistributionJobRepository distributionJobRepository;
+    @Autowired private StaticJobAccessor staticJobAccessor;
+
+    @Autowired
+    private DefaultNotificationAccessor defaultNotificationAccessor;
+
+    @AfterEach
+    public void cleanup() {
+        distributionJobRepository.deleteAll();
+    }
+
+    @Test
+    @Transactional
+    void findAndSortReturnsNotificationsOnNoFilters() {
+        long providerAndBDConfigId = 1L;  // This is also notification.providerConfigId
+
+        // Create a JobsDetailEntity and the DistributionJobEntity
+        FrequencyType frequency = FrequencyType.REAL_TIME;
+        EmailJobDetailsModel emailJobDetailsModel = createEmailJobDetails();
```

Comment:
If we can we should try to  encapsulate some of the duplicate code we find to try and cut down on the overall class size. Some of these values can either be made as private class members or just passed directly into createDistributionJobRequestModel, ex. we don't need to save an object of Frequency.REAL_TIME, we can just use  the enum directly.